pkg/morserino/com_listen.go

```diff
@@ -110,23 +123,40 @@ func Listen(port io.Reader) error {
 		}
 
 		if n == 0 {
-			fmt.Println("\nEOF")
+			AppLogger.Debug().Msg("EOF detectd")
+			AppLogger.Trace().Msg("Sending EOF to the console displayer")
+			channels.MessageBuffer <- "\nEOF"
+			//sending the exit marker to the diplay goroutine
```

Comment:
Instead of exposing your synchronization primitive, how about hiding this being an interface, something like.

```golang
type MessageWriter interface {
       WriteMessage(msg string) error
}
```

And hide the "technicalities` of how you actually send a message behind this ?
It will give you much better readability and testabillity :smile_cat: 

From the caller It will be much more easier:

```golang
_ = w.WriteMessage("\nEOF")
```